grease/src/Grease/Shape/Print.hs

```diff
@@ -94,14 +92,16 @@ evalPrinter cfg p =
   then doc
   else doc PP.<> PP.line PP.<> PP.line PP.<> printAllocs (cfgAddrWidth cfg) as
 
-printerAlloc :: Printer w (PP.Doc Void) -> Printer w BlockId
-printerAlloc computeDoc = do
+printerAlloc :: Printer w (PP.Doc Void) -> Maybe BlockId -> Printer w BlockId
+printerAlloc computeDoc bid = do
   -- The sequencing of block number allocations is important for legibility: It's
   -- nicer when pointers have lower block numbers than the pointers inside the
   -- allocation that they point to, because then allocation numbers increase
   -- while reading right-to-left, top-to-bottom.
   as <- State.get
-  let (blk@(BlockId k), as') = allocNext as
+  let (blk@(BlockId k), as') = case bid of
+        Nothing -> allocNext as
+        Just bid' -> (bid', as)
```

Comment:
This would be a great place to be able to quickly refer to a GHC-style "note" comment explaining this feature.